src/electron/device-connect-view/components/device-connect-body.tsx

```diff
@@ -67,11 +67,11 @@ export class DeviceConnectBody extends React.Component<DeviceConnectBodyProps, D
         });
     };
 
-    private OnConnectedCallback: OnConnectedCallback = (isConnected: boolean, deviceName?: string) => {
+    private OnConnectedCallback: OnConnectedCallback = (isConnected: boolean, hasFailedConnecting: boolean, deviceName?: string) => {
```

Comment:
Would an enum for connectionState be clearer than 2 bools? Something like:
connectionUnknown  (equivalent to isConnected = false, hasFailedConnecting = false)
connectionFailed (equivalent to isConnected = false, hasFailedConnection = true)
connectionValidated (equivalent to isConnected = true, hasFailedConnecting = false)

There's no state where isConnected = true and hasFailedConnecting=true

Reply:
Switched to an enum. This allowed me to eliminate an extra callback(!)